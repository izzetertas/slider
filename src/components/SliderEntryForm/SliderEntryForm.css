.input-container {
  display: flex;
  margin: 0 auto;
  padding: 40px;
  width: 90%;
  color: #333;
}

input {
  height: 40px;
  font-size: 18px;
}

input[type="text"] {
  width: 100%;
}

button {
  width: 140px;
  margin-left: 15px;
}