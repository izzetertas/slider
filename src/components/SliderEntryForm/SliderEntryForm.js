import React, { Component } from 'react'
import './SliderEntryForm.css'

class SliderEntryForm extends Component {
  state = { title: '' }

  focusToInput = () => {
    this.textInput && this.textInput.focus()
  }

  handleTitleOnChange = (e) =>  this.setState({ title : e.target.value })

  handleTitleOnKeyPress = (e) =>  e.key === 'Enter' && this.handleOnClick()

  handleOnClick = (e) => {
    const { title } = this.state
    if(!title) return

    this.setState({ title: ''})
    this.focusToInput()
    this.props.onSubmit(title)
  }

  render(){
    const { title } = this.state;

    return (
      <div className='input-container'>
        <input
          type='text'
          placeholder='Enter a title'
          autoFocus
          value={title}
          ref={input => this.textInput = input} 
          onChange={this.handleTitleOnChange}
          onKeyPress={this.handleTitleOnKeyPress}
        />
        <button onClick={this.handleOnClick}>
          Add New Slide
        </button>
      </div>
    );
  }
}

export default SliderEntryForm
