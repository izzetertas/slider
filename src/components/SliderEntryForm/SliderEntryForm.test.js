import React from 'react'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import SliderEntryForm from './SliderEntryForm'

let wrapper;

describe('<SliderEntryForm/>', () => {
  beforeEach(() => {
    wrapper = shallow(<SliderEntryForm />)
  })

  it('should render correctly', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
      expect(wrapper.find('button').length).toBe(1)
      expect(wrapper.find('input').find({type: 'text'}).length).toBe(1)
  });


  describe('when user clicks Add New Slider button', () => {
    describe('if title is empty', () => {
      it('should not call onSubmit ', () => {
        const onSubmitMock = jest.fn()
        wrapper = shallow(<SliderEntryForm onSubmit={onSubmitMock} />)
        wrapper.find('button').first().simulate('click')
        expect(wrapper.state('title')).toBe('')
        expect(onSubmitMock).toHaveBeenCalledTimes(0)
      });
    });
    
    describe('if title is not empty', () => {
      it('should call onSubmit ', () => {
        const onSubmitMock = jest.fn()
        wrapper = shallow(<SliderEntryForm onSubmit={onSubmitMock} />)
        wrapper.setState({ title: 'Test 1'})
        wrapper.find('button').first().simulate('click' )
        expect(wrapper.state('title')).toBe('')
        expect(onSubmitMock).toHaveBeenCalledWith('Test 1')
      });
    });
  });


  it.skip('should call textInput when input focuses', () => {
    wrapper = mount(<SliderEntryForm />)
    const { textInput } = wrapper.instance();
    jest.spyOn(textInput, "focus");
    wrapper.instance().componentDidMount();
    expect(textInput.focus).toHaveBeenCalledTimes(1);
  })

  describe('when user type in text input', () => {
    it('should update state correctly', () => {
      wrapper.find('input').find({type: 'text'}).first().simulate('change',  { target: { value: 'Test 1' } })
      expect(wrapper.state('title')).toEqual('Test 1')
    });
  });

  describe('when user click enter in text input', () => {
    it('should call button click method', () => {
      const onSubmitMock = jest.fn()
      wrapper = shallow(<SliderEntryForm onSubmit={onSubmitMock} />)
      wrapper.setState({ title: 'Test 1'})
      wrapper.find('input').find({type: 'text'}).first().simulate('keyPress',  { key: 'Enter' })
      expect(wrapper.state('title')).toBe('')
      expect(onSubmitMock).toHaveBeenCalledWith('Test 1')
    });
  });  
});