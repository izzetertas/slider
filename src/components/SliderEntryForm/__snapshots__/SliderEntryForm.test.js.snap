// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SliderEntryForm/> should render correctly 1`] = `
<div
  className="input-container"
>
  <input
    autoFocus={true}
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="Enter a title"
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
  >
    Add New Slide
  </button>
</div>
`;
